sequence_processing_pipeline/FastQCJob.py

```diff
@@ -94,13 +95,33 @@ def _find_projects(self, path_to_run_id_data_fastq_dir, is_raw_input):
                      'zero_files' not in x]
 
             # break files up into R1, R2, I1, I2
-            # assume _R1_ does not occur in the path as well.
-            r1_only = [x for x in files if '_R1_' in x]
-            r2_only = [x for x in files if '_R2_' in x]
-
-            # amplicon runs may or may not have an i2. this is okay.
-            i1_only = [x for x in files if '_I1_' in x]
-            i2_only = [x for x in files if '_I2_' in x]
+            # use capturing to handle both raw files as well as trimmed and
+            # filtered files. We don't need to process the captured string.
+            i1_files = compile(r"^.*_L\d{3}_I1_\d{3}\.(trimmed\.|filtered"
+                               r"\.|)fastq\.gz$")
+            i2_files = compile(r"^.*_L\d{3}_I2_\d{3}\.(trimmed\.|filtered"
+                               r"\.|)fastq\.gz$")
+            r1_files = compile(r"^.*_L\d{3}_R1_\d{3}\.(trimmed\.|filtered"
+                               r"\.|)fastq\.gz$")
+            r2_files = compile(r"^.*_L\d{3}_R2_\d{3}\.(trimmed\.|filtered"
+                               r"\.|)fastq\.gz$")
+
+            # i1_only, i2_only, r1_only, r2_only = ([] for i in range(4))
+            i1_only = []
+            i2_only = []
+            r1_only = []
+            r2_only = []
+
+            for some_path in files:
```

Comment:
Just to confirm that this is OK, the if/elif means that if a file matches more than one regex it will not be capture but only once; is that OK?

Reply:
This should be okay. Each regex begins with '^' and ends with '$', making it cover the entire filename. Given that and the structure of the regexes themselves, I don't expect a filename to match more than one.